### PR TITLE
Remove extra tracing messages from the QueryEngine

### PR DESCRIPTION
**Pull request by cprieto**
- Extra trace messages (or redundant) used for poor man profile are removed.
- OLTP tracing instrumentation macros removed
- This closes https://github.com/prisma/prisma-engines/issues/2910
- Binary size reduced by approx 3.5M in macOS
- Same instrumentation will return in a different way probably in the future

### CODE REVIEW

File: query-engine/query-engine-node-api/src/logger.rs [hunk 1]
```diff
@@ -24,10 +16,6 @@ use tracing_subscriber::{
 };
 
 pub(crate) struct Logger {
-    // We need the original parameters in case we move to tracing
-    log_queries: bool,
-    log_level: LevelFilter,
-    log_callback: ThreadsafeFunction<String>,
     __dispatcher: RwLock<Dispatch>,
```
Comment (garrensmith) [inline query-engine/query-engine-node-api/src/logger.rs:19]: Do we still need the rwlock? we not changing the dispatcher based on if tracing is happening or not. We probably can go back to creation of dispatch on instantiaion.

**Comment by garrensmith** [comment]
Do we want to keep the tracing work that runs on when we run the engine as a server? I've never used that so I'm not sure if it is useable?

File: query-engine/query-engine-node-api/src/logger.rs (hunk 1 above)
Comment (cprieto) [reply query-engine/query-engine-node-api/src/logger.rs:19]: Good finding, the `RwLock` can disapear. I will keep the Logger in its separate struct, in that way would be easier to expand it if needed in the future.

File: query-engine/query-engine-node-api/src/logger.rs [hunk 2]
```diff
@@ -24,11 +15,7 @@ use tracing_subscriber::{
 };
 
 pub(crate) struct Logger {
-    // We need the original parameters in case we move to tracing
-    log_queries: bool,
-    log_level: LevelFilter,
-    log_callback: ThreadsafeFunction<String>,
-    __dispatcher: RwLock<Dispatch>,
+    __dispatcher: Dispatch,
```
Comment (garrensmith) [inline query-engine/query-engine-node-api/src/logger.rs:18]: Sorry last thing then this should be good to go. Can you rename this back to `dispatcher`. I know its meant to be private with the `__` but that doesn't seem to be the convention https://rust-lang.github.io/api-guidelines/naming.html

File: query-engine/query-engine-node-api/src/logger.rs [hunk 3]
```diff
@@ -49,67 +36,16 @@ impl Logger {
...
-        let registry = Registry::default().with(telemetry).with(layer);
-
-        // Assign a new dispatcher now with the telemetry layer assigned
-        let mut dispatcher = self.__dispatcher.write().await;
-        *dispatcher = Dispatch::new(registry);
-
-        Ok(())
     }
 
     /// Returns a tracing dispatcher
```
Comment (garrensmith) [inline query-engine/query-engine-node-api/src/logger.rs:46]: This comment needs to be fixed as well. Its not a tracing dispatcher anymore.

**Review by garrensmith — APPROVED** [review]
Great. Thanks

**Pull request merged**